### PR TITLE
Update the logentry and metric templates with fields for monitored resources

### PR DESCRIPTION
**Pull request by ZackButcher**
Should monitored resource type be a string, or a `ValueType`? In Stackdriver it is always a string (as are all the dimensions).

**Release note**:

```release-note
Add fields for monitored resources to the metric and log entry templates.
```

---
This change is [Reviewable](https://reviewable.io/reviews/istio/mixer/1163)

### CODE REVIEW

File: template/logentry/template.proto [hunk 1]
```diff
@@ -33,4 +33,10 @@ message Template {
 
     // Severity indicates the importance of the log entry.
     string severity = 3;
+
+    // An expression to compute the type of the monitored resource this log entry is being recorded on.
+    string monitored_resource_type = 4;
```
Comment (manlinl) [inline template/logentry/template.proto]: string is too limited, e.g. Google service control uses labels, map[string]string to identify resource. 
istio.mixer.v1.config.descriptor.ValueType is more flexible. How about calling the field monitored_resource, so it doesn't have to be a resource type?

Comment (manlinl) [inline template/logentry/template.proto]: call the field monitored_resource to be generic?

File: template/metric/template.proto [hunk 2]
```diff
@@ -29,4 +29,10 @@ message Template {
 
     // The unique identity of the particular metric to report.
     map<string, istio.mixer.v1.config.descriptor.ValueType> dimensions = 2;
+
+    // An expression to compute the type of the monitored resource this metric is being reported on.
+    string monitored_resource_type = 3;
```
Comment (manlinl) [inline template/metric/template.proto]: call the field monitored_resource to be generic?

File: template/logentry/template.proto (hunk 1 above)
Comment (ZackButcher) [reply template/logentry/template.proto]: I'm not sure what you mean by generic. Today we only have a single example of monitored resources, which are in Stackdriver. Stackdriver represents a MR as a type + labels. If we call the field only "monitored_resource" then I don't think it's clear what I should populate the field with.

**Comment by ZackButcher** [comment]
PTAL

File: template/logentry/template.proto (hunk 1 above)
Comment (manlinl) [reply template/logentry/template.proto]: My understanding is the template is meant to be general purpose. So adapter can share with it. In stackdriver, it is a type + labels, in other monitor BE, could be just a url + key value pair attribute. My thought is name it resouce + dimension and let adapter developer to define its exact meaning. Not a big thing, so it is up to you.

**Comment by ZackButcher** [comment]
(Trying to poke the bot to merge things)

**Pull request merged**